src/metabase/driver.clj

```diff
@@ -940,6 +940,16 @@
   dispatch-on-initialized-driver
   :hierarchy #'hierarchy)
 
+(defmulti truncate!
+  "Delete the current contents of `table-name`.
+  If something like a SQL TRUNCATE statement is supported, we use that, but may otherwise fall back to explicitly
+  deleting rows, or dropping and recreating the table.
+  Depending on the driver, the semantics on whether triggers are fired, AUTO_INCREMENT is reset etc, is allowed to vary.
```

Comment:
Are we OK with this?

Reply:
> any pks

I'm not following. What pks are you talking about? Any uploaded `_mb_row_id` column should be ignored for appends and replacements.

Reply:
Oh, I thought we'd use them. Seems surprising that export and replace could renumber the pks, and that could play badly with a merge if they edit that exported file.

Reply:
```suggestion
  Depending on the driver, the semantics on whether triggers are fired, AUTO_INCREMENT is reset etc, are allowed to vary.
```

"the semantics…are"